crates/bcr-wdc-quote-service/src/persistence/surreal.rs

```diff
@@ -339,42 +207,60 @@ impl DBQuotes {
 #[async_trait]
 impl Repository for DBQuotes {
     async fn load(&self, qid: uuid::Uuid) -> AnyResult<Option<quotes::Quote>> {
-        self.load(qid)
-            .await?
-            .map(std::convert::TryInto::try_into)
-            .transpose()
+        let res = self.load(qid).await?.map(quotes::Quote::from);
+        Ok(res)
     }
 
-    async fn update_if_pending(&self, new: quotes::Quote) -> AnyResult<()> {
-        if matches!(new.status, quotes::QuoteStatus::Pending { .. }) {
-            return Err(anyhow!("cannot update to pending"));
-        }
-        let recordid = surrealdb::RecordId::from_table_key(&self.table, new.id);
-        self.db
-            .query("UPDATE $rid CONTENT $new WHERE status == $status")
+    async fn update_status_if_pending(
+        &self,
+        qid: uuid::Uuid,
+        new: quotes::QuoteStatus,
+    ) -> AnyResult<()> {
+        let recordid = surrealdb::RecordId::from_table_key(&self.table, qid);
+        let before: Option<DBEntryQuote> = self
+            .db
+            .query("UPDATE $rid SET status = $new WHERE status.status == $status RETURN BEFORE ")
             .bind(("rid", recordid))
-            .bind(("new", DBEntryQuote::from(new)))
-            .bind(("status", DBEntryQuoteStatus::Pending))
-            .await?;
+            .bind(("new", new))
+            .bind(("status", quotes::QuoteStatusDiscriminants::Pending))
+            .await?
+            .take(0)?;
+        if before.is_none() {
```

Comment:
I think this could be `let before = before.ok_or(anyhow::...)?` instead of the check + unwrap.
(same below)